src/modules/site/base/views/mapping.py

```diff
@@ -21,44 +29,75 @@ def mapping():
+
+    # Form info
+    'jwt_csrf_token': (get_jwt() or {}).get("csrf"),
+    'form': FileUploadForm(),
+
+    # TODO: change
+    'nemascan_container_url': 'https://github.com/AndersenLab/dockerfile/tree/nemarun/nemarun',
+    'nemascan_github_url':    'https://github.com/AndersenLab/NemaScan',
+
+    # Species list
+    'species_list': SPECIES_LIST,
+    'species_fields': [
+      'name', 'short_name', 'project_num', 'wb_ver', 'latest_release',
+    ],
+  })
 
 
 @mapping_bp.route('/mapping/upload', methods = ['POST'])
 @jwt_required()
 def submit_mapping_request():
   form = FileUploadForm(request.form)
   user = get_current_user()
+
+  # Validate form
   if not form.validate_on_submit():
     flash("You must include a description of your data and a TSV file to upload", "error")
     return redirect(url_for('mapping.mapping'))
-  
-  label = bleach.clean(request.form.get('label'))
-  props = {'label': label, 
-          'username': user.name,
-          'email': user.email,
-          'file': request.files['file']}          
+
+  # Read fields from form
+  label   = bleach.clean(request.form.get('label'))
+  species = bleach.clean(request.form.get('species'))
+
+  # Save uploaded file to server temporarily with unique generated name
+  # TODO: Is there a better way to generate this name? E.g. using a Tempfile?
+  local_path = os.path.join(uploads_dir, secure_filename(f'{ unique_id() }.tsv'))
+  request.files['file'].save(local_path)
+
+  # Package submission data together into dict
+  data = {
+    'label':    label,
+    'species':  species,
+    'filepath': local_path,
+  }
+
   try:
-    logger.warn("Check Duplicates is DISABLED !!!")
-    m = create_new_mapping(**props, check_duplicates=True)
+    # logger.warn("Check Duplicates is DISABLED !!!")
+    m = create_new_mapping(user, data, no_cache=False)
     return redirect(url_for('mapping.mapping_report', id=m.id))
+
+  except DuplicateDataError as ex:
+    flash('It looks like you submitted that data already - redirecting to your list of Mapping Reports', 'danger')
+    return redirect(url_for('mapping.mapping_user_reports'))
+
+  except CachedDataError as ex:
+    flash('It looks like that data has already been submitted - redirecting to the saved results', 'danger')
+    return redirect(url_for('mapping.mapping_report', id = ex.args[0].id))
+
   except Exception as ex:
-    if str(type(ex).__name__) == 'DuplicateDataError':
-      flash('It looks like you submitted that data already - redirecting to your list of Mapping Reports', 'danger')
-      return redirect(url_for('mapping.mapping_user_reports'))
-    if str(type(ex).__name__) == 'CachedDataError':
-      flash('It looks like that data has already been submitted - redirecting to the saved results', 'danger')
-      return redirect(url_for('mapping.mapping_report', id=ex.description))
-    logger.error(ex.description)
-    flash(f"Unable to submit your request: \"{ex.description}\"", 'danger')
+    logger.error(ex)

```

Comment:
can we combine the output of these three logger calls into one?

Reply:
Fixed in 8ef543d1e139ac211e468be13d866855c5fcd655

Reply:
thank you